src/main/java/com/denizenscript/denizencore/scripts/commands/core/MongoCommand.java

```diff
@@ -0,0 +1,765 @@
+                    && arg.asElement().isBoolean()) {
+                scriptEntry.addObject("upsert", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("action")
+                    && arg.matchesPrefix("use_database", "use_db")) {
+                scriptEntry.addObject("action", new ElementTag("use_database"));
+                scriptEntry.addObject("new_database", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("action")
+                    && arg.matchesPrefix("use_collection", "use_col")) {
+                scriptEntry.addObject("action", new ElementTag("use_collection"));
+                scriptEntry.addObject("new_collection", arg.asElement());
+            }
+            else {
+                arg.reportUnhandled();
+            }
+        }
+        if (!scriptEntry.hasObject("id")) {
+            throw new InvalidArgumentsException("Must specify an ID!");
+        }
+        if (!scriptEntry.hasObject("action")) {
+            throw new InvalidArgumentsException("Must specify an action!");
+        }
+        if (!scriptEntry.hasObject("upsert")) {
+            scriptEntry.defaultObject("upsert", new ElementTag("false"));
+        }
+    }
+
+    @Override
+    public void execute(final ScriptEntry scriptEntry) {
+        ElementTag id = scriptEntry.getElement("id");
+        ElementTag action = scriptEntry.getElement("action");
+        ObjectTag uri = scriptEntry.getObjectTag("uri");
+        ElementTag database = scriptEntry.getElement("database");
+        ElementTag collection = scriptEntry.getElement("collection");
+        ElementTag command = scriptEntry.getElement("command");
+        ObjectTag parameter = scriptEntry.getObjectTag("parameter");
+        MapTag commandMap = scriptEntry.getObjectTag("command_map");
+        MapTag findFilter = scriptEntry.getObjectTag("filter");
+        MapTag insertData = scriptEntry.getObjectTag("insert_data");
+        MapTag updateData = scriptEntry.getObjectTag("update");
+        MapTag newData = scriptEntry.getObjectTag("new");
+        ElementTag upsert = scriptEntry.getElement("upsert");
+        ElementTag findByID = scriptEntry.getElement("find_by_id");
+        ElementTag useDatabase = scriptEntry.getElement("new_database");
+        ElementTag useCollection = scriptEntry.getElement("new_collection");
+        if (scriptEntry.dbCallShouldDebug()) {
+            Debug.report(scriptEntry, getName(), id, uri, database, action);
+        }
+        try {
+            if (action.asString().equalsIgnoreCase("connect")) {
+                if (connections.containsKey(id.asString().toLowerCase())) {
+                    Debug.echoError(scriptEntry, "Already connected to a server with ID '" + id.asString() + "'!");
+                    scriptEntry.setFinished(true);
+                    return;
+                }
+                String rawUri;
+                if (uri.shouldBeType(SecretTag.class)) {

```

Comment:
You copy/pasted SecretTag references that are written weird because they have backwards-compatibility with old formats - just require a SecretTag directly, don't use this conversion logic